src/uucore/src/lib/features/memo.rs

```diff
@@ -1,22 +1,27 @@
-//! Memo runner of printf
-//! Takes a format string and arguments
-//! 1. tokenize format string into tokens, consuming
-//! any subst. arguments along the way.
-//! 2. feeds remaining arguments into function
-//! that prints tokens.
-
+//! Main entry point for our implementation of printf.
+//!
+//! The [`printf`] and [`sprintf`] closely match the behavior of the
+//! corresponding C functions: the former renders a formatted string
+//! to stdout, the latter renders to a new [`String`] object.
 use crate::display::Quotable;
-use crate::error::UResult;
-use crate::features::tokenize::sub::Sub;
-use crate::features::tokenize::token::{Token, Tokenizer};
+use crate::error::{UResult, USimpleError};
+use crate::features::tokenize::sub::SubParser;
+use crate::features::tokenize::token::Token;
 use crate::features::tokenize::unescaped_text::UnescapedText;
 use crate::show_warning;
 use itertools::put_back_n;
+use std::io::{stdout, Cursor, Write};
 use std::iter::Peekable;
 use std::slice::Iter;
 
-pub struct Memo {
-    tokens: Vec<Box<dyn Token>>,
+/// Memo runner of printf
+/// Takes a format string and arguments
+/// 1. tokenize format string into tokens, consuming
+/// any subst. arguments along the way.
+/// 2. feeds remaining arguments into function
+/// that prints tokens.
+struct Memo {
```

Comment:
I guess unrelated to this particular PR, but I don't under why this is called `Memo`? Is it like a memoization of the token string?

The same goes for the fact that `printf` and `sprintf` are exposed via the `memo` module at the moment, instead of something (imo) more appropriate like `uucore::format` or `uucore::print`

Reply:
I'm not sure why it is called `Memo`, I assumed it had something to do with memoization as well. I wasn't really familiar with what this `Memo` struct was supposed to do when I moved it from the `printf` crate to the `uucore` crate, so I just named this module `uucore::memo`, but yes it should have been something like `uucore::printf` or something like that.